### PR TITLE
Allow users to upload notifications

### PR DESCRIPTION
**Pull request by JonathanLorimer**
This PR adds support for the per-step `notify` key that buildkite accepts.

https://buildkite.com/docs/pipelines/notifications

### CODE REVIEW

**Comment by adikari** [comment]
Can you provide details of what this PR is about and what problem it solves?

**Comment by JonathanLorimer** [comment]
> Can you provide details of what this PR is about and what problem it solves?

Yes, Its only in draft right now because I was just testing it. I need to cleanup a lot of stuff (as you can see I modified the hooks executable for easier testing). 

The gist is that I want to be able to provide a custom notify clause in the monorepo-plugin config so that I can customize the github commit statuses https://buildkite.com/docs/integrations/github#customizing-commit-statuses. Right now the yaml marshalling doesn't support this key.

File: plugin.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-const pluginName = "github.com/monebag/monorepo-diff"
+const pluginName = "monorepo-diff-buildkite-plugin"
```
Comment (JonathanLorimer) [inline plugin.go:11]: I am happy to revert this change along with the `strings.Contains` below, but this seemed to be more flexible to me. Let me know what you think.

**Review by JonathanLorimer — COMMENTED** [review]
Self-review

File: plugin.go (hunk 1 above)
Comment (adikari) [reply plugin.go:11]: I prefer to leave it as it is. Once you revert it back I will merge the PR

File: plugin.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ type Step struct {
 	Env       map[string]string `yaml:"env,omitempty"`
 	Async     bool              `yaml:"async,omitempty"`
 	SoftFail  interface{}       `json:"soft_fail" yaml:"soft_fail,omitempty"`
+  Notify    interface{}       `yaml:"notify,omitempty"`
```
Comment (adikari) [inline plugin.go:56]: Is it possible to add proper type with possible values? Will serve as a documentation and also validation.
